From e57539ae17c0e452fd9fe88f0953b27fd3e4dbb1 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Tue, 30 Oct 2018 14:57:24 +0100
Subject: [PATCH 05/60] build: gcc-8: disambiguate SnapContainerData type spec
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

../../third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h:104:3: error: declaration of ‘base::Optional<cc::SnapContainerData> blink::ScrollPaintPropertyNode::SnapContainerData() const’ [-fpermissive]
   }
   ^
In file included from ../../third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h:17,
                 from ../../third_party/blink/renderer/platform/graphics/paint/transform_paint_property_node.h:13,
                 from ../../third_party/blink/renderer/platform/graphics/paint/clip_paint_property_node.h:13,
                 from ../../third_party/blink/renderer/core/paint/object_paint_properties.h:15,
                 from ../../third_party/blink/renderer/core/paint/fragment_data.h:9,
                 from ../../third_party/blink/renderer/core/layout/layout_object.h:46,
                 from ../../third_party/blink/renderer/core/layout/svg/layout_svg_model_object.h:34,
                 from ../../third_party/blink/renderer/core/layout/svg/layout_svg_container.h:26,
                 from ../../third_party/blink/renderer/core/layout/svg/layout_svg_hidden_container.h:23,
                 from ../../third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.h:23,
                 from ../../third_party/blink/renderer/core/layout/svg/layout_svg_resource_paint_server.h:23,
                 from ../../third_party/blink/renderer/core/layout/svg/layout_svg_resource_paint_server.cc:23:
../../third_party/blink/renderer/platform/scroll/scroll_snap_data.h:22:48: error: changes meaning of ‘SnapContainerData’ from ‘using SnapContainerData = class cc::SnapContainerData’ [-fpermissive]
 using SnapContainerData = cc::SnapContainerData;
---
 .../platform/graphics/paint/scroll_paint_property_node.h        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h b/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
index 35f833733e6c..5c5545b54076 100644
--- a/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
+++ b/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
@@ -105,7 +105,7 @@ class PLATFORM_EXPORT ScrollPaintPropertyNode
     return state_.overscroll_behavior.y;
   }
 
-  base::Optional<SnapContainerData> GetSnapContainerData() const {
+  base::Optional<cc::SnapContainerData> GetSnapContainerData() const {
     return state_.snap_container_data;
   }
 
-- 
2.19.1

